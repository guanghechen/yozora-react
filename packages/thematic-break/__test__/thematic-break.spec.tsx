import { mount, render } from 'enzyme'
import React from 'react'
import ThematicBreak from '../src'

describe('prop types', function () {
  beforeEach(() => {
    jest.spyOn(global.console, 'error').mockImplementation((...args) => {
      throw new Error(args.join(' '))
    })
  })

  it('forward ref', function () {
    const ref = React.createRef<HTMLHRElement>()
    const wrapper = mount(
      <ThematicBreak ref={ref} data-name="yozora-thematic-break" />,
    )

    const o = wrapper.getDOMNode()
    expect(o).toEqual(ref.current)
    expect(o.getAttribute('data-name')).toEqual('yozora-thematic-break')
  })

  it('className is optional', function () {
    expect(render(<ThematicBreak />).hasClass('yozora-thematic-break')).toEqual(
      true,
    )

    expect(
      render(<ThematicBreak className="my-thematic-break" />).hasClass(
        'my-thematic-break',
      ),
    ).toEqual(true)
  })
})

describe('snapshot', function () {
  it('default', function () {
    const wrapper = render(<ThematicBreak />)
    expect(wrapper).toMatchSnapshot()
  })

  it('custom', function () {
    const wrapper = render(
      <ThematicBreak
        className="custom-class"
        data-name="yozora-thematic-break"
        style={{ color: 'orange' }}
      />,
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('children is not allowed', function () {
    const wrapper = render(
      <ThematicBreak
        className="custom-class"
        data-name="yozora-thematic-break"
        style={{ color: 'orange' }}
      >
        <em>Children Content will be ignored.</em>
      </ThematicBreak>,
    )
    expect(wrapper).toMatchSnapshot()
  })
})
