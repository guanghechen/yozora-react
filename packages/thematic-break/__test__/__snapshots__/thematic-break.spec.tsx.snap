// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot children is not allowed 1`] = `
<hr
  class="custom-class"
  data-name="yozora-thematic-break"
  style="color:orange"
/>
`;

exports[`snapshot custom 1`] = `
<hr
  class="custom-class"
  data-name="yozora-thematic-break"
  style="color:orange"
/>
`;

exports[`snapshot default 1`] = `
<hr
  class="yozora-thematic-break"
/>
`;
